### PR TITLE
[React utils refactor] inline code that won't live in new packages

### PR DESCRIPTION
**Pull request by benoitgrelard**

### CODE REVIEW

File: packages/react/utils/src/index.ts [hunk 1]
```diff
@@ -2,13 +2,10 @@ export * from './createContext';
 export * from './extendComponent';
 export * from './useCallbackRef';
 export * from './useComposedRefs';
-export * from './useConstant';
 export * from './useControlledState';
 export * from './useDisableBodyPointerEvents';
 export * from './useDocumentRef';
-export * from './useId';
```
Comment (benoitgrelard) [inline packages/react/utils/src/index.ts:9]: fully replaced by `@radix-ui/react-id`

File: packages/react/scroll-area/src/ScrollArea.tsx [hunk 2]
```diff
@@ -1865,6 +1863,41 @@ function useBorderBoxResizeObserver(
   }, [onResize, ref]);
 }
 
+function usePrefersReducedMotion(nodeRef: React.RefObject<Element>) {
```
Comment (benoitgrelard) [inline packages/react/scroll-area/src/ScrollArea.tsx:1866]: Simplified and inlined

File: packages/react/scroll-area/src/ScrollArea.tsx [hunk 3]
```diff
@@ -1865,6 +1863,41 @@ function useBorderBoxResizeObserver(
...
+  }, [nodeRef]);
+
+  return prefersReducedMotion;
+}
+
+/**
+ * React hook for creating a value exactly once.
+ * @see https://github.com/Andarist/use-constant
+ */
+function useConstant<ValueType>(fn: () => ValueType): ValueType {
```
Comment (benoitgrelard) [inline packages/react/scroll-area/src/ScrollArea.tsx:1893]: inlined as is

File: packages/react/scroll-area/src/ScrollArea.tsx [hunk 4]
```diff
@@ -1865,6 +1863,41 @@ function useBorderBoxResizeObserver(
...
+        setPrefersReducedMotion(!event.matches);
+      }
+      const mediaQueryList = globalWindow.matchMedia('(prefers-reduced-motion: no-preference)');
+      mediaQueryList.addEventListener('change', handleChange);
+      return function () {
+        mediaQueryList.removeEventListener('change', handleChange);
+      };
+    } catch (e) {}
+
+    return function () {};
```
Comment (jjenzz) [inline packages/react/scroll-area/src/ScrollArea.tsx:1883]: I know this wasn't you but if we're simplifying, why do we need the `try`/`catch` and the empty `return function () {}` here?

```suggestion
    function handleChange(event: MediaQueryListEvent) {
      setPrefersReducedMotion(!event.matches);
    }
    const mediaQueryList = globalWindow.matchMedia('(prefers-reduced-motion: no-preference)');
    mediaQueryList.addEventListener('change', handleChange);
    
    return () => {
      mediaQueryList.removeEventListener('change', handleChange);
    };
```

**Review by jjenzz — APPROVED** [review]

File: packages/react/scroll-area/src/ScrollArea.tsx (hunk 4 above)
Comment (benoitgrelard) [reply packages/react/scroll-area/src/ScrollArea.tsx:1883]: My guess is he was trying to get around the fact that some browsers don't support this. But given this issue #500 I'd say it doesn't really work as this is the error browsers that don't support this throw:

error

In fact, I don't really understand why the try catch doesn't catch that :think

**Pull request merged**